package/linux/dfhack

```diff
@@ -68,12 +68,37 @@ setarch_arch=$(cat hack/dfhack_setarch.txt || printf i386)
 case "$1" in
   -g | --gdb)
     shift
+    echo "set exec-wrapper env LD_LIBRARY_PATH='$LD_LIBRARY_PATH' LD_PRELOAD='$PRELOAD_LIB' MALLOC_PERTURB_=45" > gdbcmd.tmp
```

Comment:
Is this the same as `startup-with-shell off` below? I remember `startup-with-shell` being added recently to fix an issue, so I would like to stick with it unless they're equivalent.

Reply:
exec-wrapper runs inside startup shell setting up environment. It improves argument handling because gdb argument handling without shell doesn't support spaces.  df -gen parameter is likely to require a parameter with a spaces. That why keeping shell and using exec-wrapper helps if someone wants to use parameters and gdb.

Too bad gdbserver connection doesn't handle spaces or exec-wrapper so my prefered debug setup won't handle arguments.

Reply:
#1280 is what I was referring to. So you've checked that `-g` still works with your changes?

Reply:
It works. I rechecked it now:
./dfhack -g -gen 4 4 "POCKET ISLAND"
That opened df with dfhack loaded.

Reply:
It's really the GDB part I'm concerned about, but thanks for double-checking!